src/bin/simple_attachment.rs

```diff
@@ -0,0 +1,175 @@
+            OnlinePlayground::new(
+                ditto_root,
+                app_id,
+                shared_token,
+                enable_cloud_sync,
+                custom_auth_url,
+            )
+        })?
+        .build()?;
+
+    // Begin sync, then open the Ditto Store so we can insert or query documents
+    ditto.start_sync()?;
+    let store = ditto.store();
+
+    match cli.cmd {
+        Cmd::UploadPhoto { path } => {
+            upload_photo(store, &path).await?;
+        }
+        Cmd::DownloadPhoto { name } => {
+            download_photo(store, &name).await?;
+        }
+    }
+
+    Ok(())
+}
+
+async fn upload_photo(store: &Store, path: &Path) -> Result<()> {
+    let photo_name = path.file_name().and_then(|s| s.to_str()).unwrap_or("photo");
+
+    let photo_attachment = store.new_attachment(path, HashMap::default()).await?;
+    let _result = store
+        .execute(
+            "INSERT INTO COLLECTION photos (photo_attachment ATTACHMENT) DOCUMENTS (:photo_doc)",
+            Some(
+                serde_json::json!({
+                    "photo_doc": {
+                        "photo_name": photo_name,
+                        "photo_attachment": photo_attachment
+                    }
+                })
+                .into(),
+            ),
+        )
+        .await?;
+
+    println!("Uploaded photo with name '{photo_name}'");
+    Ok(())
+}
+
+async fn download_photo(store: &Store, name: &str) -> Result<()> {
+    let result = store
+        .execute(
+            "SELECT * FROM COLLECTION photos (photo_attachment ATTACHMENT)",
+            Some(
+                serde_json::json!({
+                    "photo_name": name
+                })
+                .into(),
+            ),
+        )
+        .await?;
+
+    let result_item = result
+        .get_item(0)
+        .context("result set contained no items")?
+        .value();
+
+    let photo_attachment = result_item
+        .get("photo_attachment")
+        .context("failed to find photo_attachment")?;
+
+    let photo_attachment_token = photo_attachment
+        .as_object()
+        .context("failed to get attachment token")?;
+
+    let photo_id = photo_attachment
+        .get("id")
+        .context("failed to get ID of attachment")?
+        .clone(); // Cloned to move into closure below
+
+    let _fetcher = store.fetch_attachment(photo_attachment_token, move |event| {
+        use DittoAttachmentFetchEvent::*;
+        match event {
+            Progress {
+                downloaded_bytes,
+                total_bytes,
+            } => {
+                println!("Fetcher progress for attachment {photo_id:?}: {downloaded_bytes}b/{total_bytes}b");
+            }
+            Completed { attachment } => {
+                println!("Successfully downloaded attachment {photo_id:?} to path {}", attachment.path().display());
+            }
+            Deleted => unreachable!("attachment should not get deleted while fetching"),
```

Comment:
```suggestion
            Deleted => panic!("attachment was deleted while fetching"),
```

Technically somebody using the portal or whatnot might delete the attachment right under our feet, so I wouldn't deem that `unreachable!`, even if we can deem it `panic!`-worthy; WDYT?

Reply:
Sounds good to me!